### PR TITLE
Fix assert when moving tracks

### PR DESCRIPTION
**Pull request by pierricgimmig**
Use visible_tracks_ iterator before it is invalidated by insertion.

### CODE REVIEW

**Review by reichlfl — APPROVED** [review]
LGTM, thanks!

File: OrbitGl/TrackManager.cpp [hunk 1]
```diff
@@ -252,8 +252,8 @@ void TrackManager::UpdateMovingTrackSorting() {
     int moving_track_current_position = -1;
     for (auto track_it = visible_tracks_.begin(); track_it != visible_tracks_.end(); ++track_it) {
       if (moving_track->GetPos()[1] >= (*track_it)->GetPos()[1]) {
-        visible_tracks_.insert(track_it, moving_track);
```
Comment (reichlfl) [inline OrbitGl/TrackManager.cpp:255]: I already LGTM'd this, but afterwards, I was just wondering: Wouldn't it be clearer to make this

```
auto inserted_it = visible_tracks_.insert(track_it, moving_track);
moving_track_current_position = inserted_it - visible_tracks_.begin();
```

Comment (pierricgimmig) [reply OrbitGl/TrackManager.cpp:255]: Yep lgtm, I made the change, thanks!

**Comment by karupayun** [comment]
Thanks again!

**Pull request merged**